src/StatusQ/Popups/StatusModal.qml

```diff
@@ -0,0 +1,72 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14 as QC
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+
+import "statusModal" as Spares
+
+QC.Popup {
+    id: statusModal
+
+    property Component content
+
+    property alias showBack: footerImpl.showBack
+
+    property StatusModalHeaderSettings header: StatusModalHeaderSettings {}
+    property alias buttons: footerImpl.buttons
+
+    signal backClicked()
+    signal butttonClicked(var buttonIndex)
+    signal detailsClicked()
+    signal editClicked()
+
+    parent: QC.Overlay.overlay
+
+    width: 480
+    height: contentItem.implicitHeight
+
+    margins: 0
+    padding: 0
+
+    modal: true
+
+
+    background: Rectangle {
+        color: Theme.palette.indirectColor1
+        radius: 6
+    }
+
+    contentItem: Column {
+        width: parent.width
+        Spares.StatusModalHeader {
+            id: headerImpl
+            width: 480
+
+            title: header.title
+            subTitle: header.subTitle
+            image: header.image
+            detailsButtonSettings: header.detailsButtonSettings
+
+            onEditClicked: statusModal.editClicked()
+            onDetailsClicked: statusModal.detailsClicked()
+            onClose: statusModal.close()
+        }
+
+        Loader {
+            active: true
+            anchors.horizontalCenter: parent.horizontalCenter
+            sourceComponent: statusModal.content
+        }
+
+        Spares.StatusModalFooter {
+            id: footerImpl
+            width: 480
+
+            onBack: statusModal.backClicked()
+            onClicked: statusModal.butttonClicked(buttonIndex)
```

Comment:
So, if we keep the API open, that people can do:

```
(footer).(right)Buttons: [
  StatusButton { ... }
]
```

People can just attach `onClicked` handlers directly on their buttons, so there's no need to emit  an `buttonClicked(index)` event